packages/cli/src/commands/hydrogen/init.test.ts

```diff
@@ -1,188 +1,475 @@
-    }));
-    vi.mock('@shopify/cli-kit/node/node-package-manager', async () => {
-      const original = await vi.importActual<
-        typeof import('@shopify/cli-kit/node/node-package-manager')
-      >('@shopify/cli-kit/node/node-package-manager');
-
-      return {
-        ...original,
-        installNodeModules: vi.fn(),
-        getPackageManager: () => Promise.resolve('npm'),
-      };
-    });
-    vi.mocked(outputContent).mockImplementation(() => ({
-      value: '',
-    }));
-    vi.mock('@shopify/cli-kit/node/ui', async () => {
-      const original = await vi.importActual<
-        typeof import('@shopify/cli-kit/node/ui')
-      >('@shopify/cli-kit/node/ui');
-
-      return {
-        ...original,
-        renderConfirmationPrompt: vi.fn(),
-        renderSelectPrompt: vi.fn(),
-        renderTextPrompt: vi.fn(),
-        renderInfo: vi.fn(),
-      };
-    });
-    vi.mock('@shopify/cli-kit/node/fs', async () => ({
-      fileExists: () => Promise.resolve(true),
-      isDirectory: () => Promise.resolve(false),
-      copyFile: () => Promise.resolve(),
-      rmdir: () => Promise.resolve(),
-    }));
+    vi.clearAllMocks();
+    outputMock.clear();
   });
 
-  const defaultOptions = (stubs: Record<any, unknown>) => ({
-    language: 'js' as const,
-    path: 'path/to/project',
-    ...stubs,
+  it('checks Hydrogen version', async () => {
+    await temporaryDirectoryTask(async (tmpDir) => {
+      const showUpgradeMock = vi.fn((param?: string) => ({
+        currentVersion: '1.0.0',
+        newVersion: '1.0.1',
+      }));
+      vi.mocked(checkHydrogenVersion).mockResolvedValueOnce(showUpgradeMock);
+      vi.mocked(handleProjectLocation).mockResolvedValueOnce(undefined);
+
+      const project = await runInit({path: tmpDir, git: false});
+
+      expect(project).toBeFalsy();
+      expect(checkHydrogenVersion).toHaveBeenCalledOnce();
+      expect(showUpgradeMock).toHaveBeenCalledWith(
+        expect.stringContaining('npm create @shopify/hydrogen@latest'),
+      );
+    });
   });
 
-  describe.each([
-    {
-      flag: 'installDeps',
-      value: true,
-      condition: {fn: renderConfirmationPrompt, match: /install dependencies/i},
-    },
-    {
-      flag: 'language',
-      value: 'ts',
-      condition: {fn: renderSelectPrompt, match: /language/i},
-    },
-    {
-      flag: 'path',
-      value: './my-app',
-      condition: {fn: renderTextPrompt, match: /where/i},
-    },
-  ])('flag $flag', ({flag, value, condition}) => {
-    it.skip(`does not prompt the user for ${flag} when a value is passed in options`, async () => {
+  describe('local templates', () => {
+    it('creates basic projects', async () => {
       await temporaryDirectoryTask(async (tmpDir) => {
-        // Given
-        const options = defaultOptions({
+        await runInit({
           path: tmpDir,
-          [flag as string]: value,
+          git: false,
+          language: 'ts',
+          mockShop: true,
```

Comment:
```suggestion
          mockShop: true,
          routes: false,
```

Nit, but I prefer being explicit with all the options passed in.

Reply:
The thing is, `undefined` is a valid value because that means you're not passing `--routes` flag and it will trigger a prompt. If we pass `routes: boolean` always then it doesn't run the prompt code.